### PR TITLE
Create external-identity-policy.md

### PR DESCRIPTION
**Pull request by KuiGithui**
Schema update PR: 5549878

### CODE REVIEW

**Review by bashirkunya — APPROVED** [review]

File: api-reference/beta/api/externalidentitiespolicy-update.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+```
+
+## Request headers
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}. Required.|
+|Content-Type|application/json. Required.|
+
+## Request body
+[!INCLUDE [In the request body, supply the values for relevant fields that should be updated. Existing properties that are not included in the request body will maintain their previous values or be recalculated based on changes to other property values. For best performance, don't include existing values that haven't changed.](../../includes/update-property-table-intro.md)]
```
Comment (bashirkunya) [inline api-reference/beta/api/externalidentitiespolicy-update.md:43]: This text should not be in brackets

File: api-reference/beta/resources/externalidentitiespolicy.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+---
+title: "externalIdentitiesPolicy resource type"
+description: "Resources for External Identities Policy"
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/externalidentitiespolicy.md:3]: Ideally, could be similar to the first sentence of the description below. Also good for SEO.
```suggestion
description: "Represents the tenant-wide policy that controls whether external users can leave an Azure AD tenant via self-service controls."
```

File: api-reference/beta/resources/externalidentitiespolicy.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+doc_type: resourcePageType
+---
+
+# externalIdentitiesPolicy resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Gives tenant administrator control over external identities by allowing or restricting external identities to leave an organization via self-service controls. When permitted by the administrator, external identities will have access to self-service controls via the My Account portal.
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/externalidentitiespolicy.md:16]: Suggestion to be explicit about what these self-service capabilities are/allowed them to do
```suggestion
Represents the tenant-wide policy that controls whether external users can leave the guest Azure AD tenant via self-service controls. When permitted by the administrator, external users can leave the guest Azure AD tenant through the **organizations** menu of the [My Account](https://myaccount.microsoft.com/) portal.
```

File: api-reference/beta/resources/externalidentitiespolicy.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+|[Get externalIdentitiesPolicy](../api/externalidentitiespolicy-get.md)|[externalIdentitiesPolicy](../resources/externalidentitiespolicy.md)|Read the properties and relationships of an [externalIdentitiesPolicy](../resources/externalidentitiespolicy.md) object.|
+|[Update externalIdentitiesPolicy](../api/externalidentitiespolicy-update.md)|[externalIdentitiesPolicy](../resources/externalidentitiespolicy.md)|Update the properties of an [externalIdentitiesPolicy](../resources/externalidentitiespolicy.md) object.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|allowDeletedIdentitiesDataRemoval|Boolean|The setting will inform AAD that cleanup of the external identities user information stored within the local tenant should be deleted (or not) when the user is deleted in their home directory.|
+|allowExternalIdentitiesToLeave|Boolean|The setting will inform AAD whether allow or restrict external identities to leave the tenant.If set to false, self service is not enabled, and therefore admin will need to manually perform the clean up.|
+|displayName|String|Contains string data for policy display information. Inherited from [policyBase](../resources/policybase.md).|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/externalidentitiespolicy.md:31]:
```suggestion
|allowDeletedIdentitiesDataRemoval|Boolean|Notifies Azure AD whether to clean up the user information about the external identity, from the guest tenant, when the user is deleted in their home tenant. |
|allowExternalIdentitiesToLeave|Boolean|Defines whether external users can leave the guest tenant. If set to `false`, self-service controls are not enabled, and the admin of the guest tenant must manually remove the external user from the guest tenant.|
|displayName|String|The policy name. Inherited from [policyBase](../resources/policybase.md).|
```

File: api-reference/beta/resources/externalidentitiespolicy.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+}
+-->
+``` json
+{
+"id": "String (identifier)",
+"deletedDateTime":null,
+"allowExternalIdentitiesToLeave": true,
+"allowDeletedIdentitiesDataRemoval": false,
+"displayName": "String"
+}
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/externalidentitiespolicy.md:53]: The JSON representation has data types and shouldn't be confused with a response object.
Added the properties inherited from policyBase and upstream resources/base types.
```suggestion
{
  "@odata.type": "#microsoft.graph.externalIdentitiesPolicy",
  "id": "String (identifier)",
  "deletedDateTime": "String (timestamp)",
  "description": "String",
  "displayName": "String",
  "allowExternalIdentitiesToLeave": "Boolean",
  "allowDeletedIdentitiesDataRemoval": "Boolean"
}
```

File: api-reference/beta/api/externalidentitiespolicy-get.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
+---
+title: "Get externalIdentitiesPolicy"
+description: "Read the properties and relationships of an externalIdentitiesPolicy object."
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-get.md:3]: Suggestion to make it user/SEO friendly
```suggestion
description: "Read the properties and relationships of the tenant-wide externalIdentitiesPolicy object that controls whether external users can leave an Azure AD tenant via self-service controls."
```

File: api-reference/beta/api/externalidentitiespolicy-get.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+ms.prod: "identity-and-sign-in"
+doc_type: apiPageType
+---
+
+# Get externalIdentitiesPolicy
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of an [externalIdentitiesPolicy](../resources/externalidentitiespolicy.md) object.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-get.md:15]:
```suggestion
Read the properties and relationships of the tenant-wide [externalIdentitiesPolicy](../resources/externalidentitiespolicy.md) object that controls whether external users can leave an Azure AD tenant via self-service controls.
```

File: api-reference/beta/api/externalidentitiespolicy-get.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+<!-- {
+  "blockType": "ignored"
+}
+-->
+``` http
+GET /policies/externalIdentitiesPolicy
+```
+
+## Optional query parameters
+This method supports some of the OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-get.md:37]: I see query parameters are marked as unsupported in your API.md
```suggestion
This method does not support the OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).
```

File: api-reference/beta/api/externalidentitiespolicy-get.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+## Examples
+
+### Request
+<!-- {
+  "blockType": "request",
+  "name": "get_externalidentitiespolicy"
+}
+-->
+``` http
+GET https://graph.microsoft.com/v1.0/policies/externalidentitiespolicy
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-get.md:60]:
```suggestion
GET https://graph.microsoft.com/beta/policies/externalIdentitiesPolicy
```

File: api-reference/beta/api/externalidentitiespolicy-get.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  "@odata.type": "microsoft.graph.externalIdentitiesPolicy"
+}
+-->
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "value": {
+    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#policies/externalIdentitiesPolicy/$entity",
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-get.md:78]:
```suggestion
    "@odata.context": "https://graph.microsoft.com/beta/$metadata#policies/externalIdentitiesPolicy/$entity",
```

File: api-reference/beta/api/externalidentitiespolicy-update.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
+---
+title: "Update externalIdentitiesPolicy"
+description: "Update the properties of an externalIdentitiesPolicy object."
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-update.md:3]:
```suggestion
description: "Update the settings of the tenant-wide externalIdentitiesPolicy object that controls whether external users can leave an Azure AD tenant via self-service controls."
```

File: api-reference/beta/api/externalidentitiespolicy-update.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+ms.prod: "identity-and-sign-in"
+doc_type: apiPageType
+---
+
+# Update externalIdentitiesPolicy
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Update the properties of an [externalIdentitiesPolicy](../resources/externalidentitiespolicy.md) object.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-update.md:15]:
```suggestion
Update the settings of the tenant-wide [externalIdentitiesPolicy](../resources/externalidentitiespolicy.md) object that controls whether external users can leave an Azure AD tenant via self-service controls.
```

File: api-reference/beta/api/externalidentitiespolicy-update.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+|Content-Type|application/json. Required.|
+
+## Request body
+
+[!INCLUDE [table-intro](../../includes/update-property-table-intro.md)]
+
+|Property|Type|Description|
+|:---|:---|:---|
+|allowDeletedIdentitiesDataRemoval|Boolean|The setting will inform AAD that cleanup of the external identities user information stored within the local tenant should be deleted (or not) when the user is deleted in their home directory. Required.|
+|allowExternalIdentitiesToLeave|Boolean|The setting will inform AAD whether allow or restrict external identities to leave the tenant.If set to false, self service is not enabled, and therefore admin will need to manually perform the clean up. Required.|
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-update.md:49]: Aligning with the descriptions in the properties table
```suggestion
|allowDeletedIdentitiesDataRemoval|Boolean|Notifies Azure AD whether to clean up the user information about the external identity, from the guest tenant, when the user is deleted in their home tenant. Required.|
|allowExternalIdentitiesToLeave|Boolean|Notifies Azure AD whether to clean up the user information about the external identity, from the guest tenant, when the user is deleted in their home tenant. Required.|
```

File: api-reference/beta/api/externalidentitiespolicy-update.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+## Response
+
+If successful, this method returns a `204 No Content` response code. It does not return anything in the response body.
+
+## Examples
+## Example 1: Update or set external identity not to leave tenant.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-update.md:56]: You have only 1 example, so no need for this.
```suggestion
```

File: api-reference/beta/api/externalidentitiespolicy-update.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+### Request
+<!-- {
+  "blockType": "request",
+  "name": "update_externalidentitiespolicy"
+}
+-->
+``` http
+PATCH https://graph.microsoft.com/v1.0/policies/externalidentitiespolicy
```
Comment (FaithOmbongi) [inline api-reference/beta/api/externalidentitiespolicy-update.md:65]:
```suggestion
PATCH https://graph.microsoft.com/beta/policies/externalIdentitiesPolicy
```

File: changelog/Microsoft.DirectoryServices.json [hunk 16]
```diff
@@ -1,5 +1,31 @@
 {
   "changelog": [
+    {
+      "ChangeList": [
+        {
+          "Id": "6c5c31e1-51bb-45c2-b769-4928b4977187",
+          "ApiChange": "Resource",
+          "ChangedApiName": "externalIdentitiesPolicy",
+          "ChangeType": "Addition",
+          "Description": "Added the [externalIdentitiesPolicy](https://docs.microsoft.com/en-us/graph/api/resources/externalIdentitiesPolicy?view=graph-rest-beta) resource",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:10]: Recommending a little more info to the customer.
```suggestion
          "Description": "Added the [externalIdentitiesPolicy](https://docs.microsoft.com/en-us/graph/api/resources/externalIdentitiesPolicy?view=graph-rest-beta) resource and associated methods. The external identity policy represents the tenant-wide policy that controls whether external users can leave an Azure AD tenant via self-service controls.",
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @KuiGithui for working on this PR. I've recommended some minor fixes. The only gap:
- Please document the **externalIdentitiesPolicy** relationship in [policyRoot](https://docs.microsoft.com/en-us/graph/api/resources/policyroot?view=graph-rest-beta).

And we'll be good to go.

File: api-reference/beta/api/externalidentitiespolicy-update.md (hunk 1 above)
Comment (FaithOmbongi) [reply api-reference/beta/api/externalidentitiespolicy-update.md]: +1, this was the cause of the validation error.

**Pull request merged**